backends/graphics/surfacesdl/surfacesdl-graphics.cpp

```diff
@@ -1550,17 +1550,17 @@ bool SurfaceSdlGraphicsManager::saveScreenshot(const Common::Path &filename) con
 
 	SDL_Palette *sdlPalette = _hwScreen->format->palette;
 	if (sdlPalette) {
-		byte palette[256 * 3];
+		Graphics::Palette palette(256);
 		for (int i = 0; i < sdlPalette->ncolors; i++) {
-			palette[(i * 3) + 0] = sdlPalette->colors[i].r;
-			palette[(i * 3) + 1] = sdlPalette->colors[i].g;
-			palette[(i * 3) + 2] = sdlPalette->colors[i].b;
+			palette.data[(i * 3) + 0] = sdlPalette->colors[i].r;
```

Comment:
I'm wondering whether `palette` shouldn't allow setting also individual R/G/B components to avoid implementation-dependant code like this?

Reply:
Maybe this could be simplified in one call, something like `palette.setRGB(r, g, b)` as loops like this tend to set all three at once.

Reply:
Functions to set and get an entry by index makes sense as uses of setPalette and grabPalette elsewhere do work with a single entry at a time.
I do not think we should protect the data at this point.

Reply:
Added the single-entry set/get.